### ISSUE TITLE
Support for Monoprice 4 Zone Amp?

### ISSUE THREAD

**Issue by CaptainStealthy** [etsinko/pymonoprice#8]
Hi @etsinko, I sincerely apologize for misusing this repo for this question, but I didn't want pollute the [home-assistant/core](https://github.com/home-assistant/core) issues list, and I couldn't find another way to contact you or @OnFreund.

I was just wondering if your Monoprice integration for Home Assistant supports the [4 zone amplifier (44519)](https://www.monoprice.com/product?p_id=44519) as well? Or is it just the 6 zone version?

The reason I ask is because I am looking at whole home audio solutions for my new house, and while the 6 zone amp is good bang for the buck, the 4 zone amp provides more power to each zone (60W vs 25W at 8 ohms).

I am confirming whether the speakers we're going to install can actually support/use the extra power, but if they can, then I would prefer the extra power that comes with the 4 zone amp. But I need to make sure that it'll still work with HA, lol.

Any light either of you can shed on the subject would be appreciated. :)

Thanks!

**Comment by OnFreund** [comment]
At a first glance the serial API looks similar, but I haven't really dug into it. Keep in mind that without testing we will never know for sure.

**Comment by CaptainStealthy** [comment]
Thanks so much for the quick response! And yeah, I get that you'd need to actually test with that amp to know for sure.

I still have a lot of time before I actually have to buy anything, so if you get a chance to dig into it further, anything else you can find out would be appreciated! 🙂

**Comment by OnFreund** [comment]
It'll take me some time to dig deep. If you want to speed things up, you can look at the RS232 protocol description in both manuals and see how different they are (if at all).

**Comment by CaptainStealthy** [comment]
Fair enough - the info in the manuals may go over my head, but I'll take a look and see what I can decipher from it, and post back here with what I find, if that works.

Thanks!

**Comment by CaptainStealthy** [comment]
@OnFreund I obviously have nothing better to do tonight (lol)... so I took a quick look through the RS232 pages in both manuals.

The only differences I can see is in the `Inquiry command structure (2)` sections. The 6 zone amp has values of `15` and `16` listed (for zones 5 and 6) for the control object code, whereas the 4 zone amp does not - for obvious reasons.

Everything else, from what my late night brain can see, looks identical. But I don't have the knowledge or experience to say whether it's that simple, or if you guys would need to make other modifications to support the other amp. And like you said, no one can say for sure without physically testing the 4 zone amp.

Hopefully that helps... feel free to let me know if there is anything else I can do to help. :)

**Comment by CaptainStealthy** [comment]
(sorry, had this typed out, but never sent it!)

Sounds good - it's likely going to be a while, and I still haven't narrowed down my list and finalized which amp I plan to get. But this 4 zone amp from Monoprice is the frontrunner, because it provides the most amount of power (60W per speaker) for a very reasonable price.

I'll keep you posted, though. Obviously I'd love to control the amp with HA, but if I can't, given its price compared to the competition, it may not be the end of the world, especially if I can use something like an IR blaster to send commands to it.

**Comment by CaptainStealthy** [comment]
Hey guys,

Just commenting to say that I will eventually be looking at this - probably not until at least September, though.

It might be sooner depending on whether I'm able to buy the amp before we move to the new house - or if my wife decides she wants full-home audio working on the day we move in... 👀 

But as you can imagine, the priority (both time/effort, as well as money, lol) are on other things at the moment. 😆

**Comment by CaptainStealthy** [comment]
Update - Finally moved into the new house...significantly behind schedule, but we got there, lol

I'm still planning to get the Monoprice amp - tempted because it's on sale right now, but I think I need to wait a little longer due to...conflicting fiscal priorities, haha.

I'll make sure to comment again once I have the amp and can test - just wanted to make sure the thread doesn't go stale :)
